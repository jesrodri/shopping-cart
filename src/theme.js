const colors = {
  black:       '#000000',
  white:       '#ffffff',
  lightGray:   '#b5b0a7',
  gray:        '#999999',
  darkGray:    '#37352f',
  darkRed:     '#cc0000',
  yellow:      '#e0ca00',
  lightYellow: '#f7e581',
  salmon:      '#fa8072',
  darkSalmon:  '#f16050',
  green:       '#66f57e',
  darkGreen:   '#49a659',
}

const space = [0, 4, 8, 16, 32, 64, 128, 256, 512]

const fonts = {
  body:
    "-apple-system, BlinkMacSystemFont, 'Segoe UI', Helvetica, 'Apple Color Emoji', Arial, sans-serif, 'Segoe UI Emoji', 'Segoe UI Symbol', 'Nunito', sans-serif;",
}

const fontSizes = [0, 5, 7, 9, 12, 16, 21, 28, 37, 50, 67, 89, 119, 159].map(
  (n) => `${n / 10}rem`
)

const fontWeights = [0, 300, 400, 700]

const radii = [0, 2]

const zIndices = []

const sizes = []

const theme = {
  colors,
  space,
  fonts,
  fontSizes,
  fontWeights,
  radii,
  zIndices,
  sizes,
}

export default theme
