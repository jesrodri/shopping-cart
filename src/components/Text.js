import styled from "styled-components";
import theme from "../theme";

const Text = styled.text`
  display: block;
  margin-block-end: ${({ theme }) => theme.space[4]};
  line-height: ${({ theme }) => theme.space[5]};
  text-align: ${(props) => props.align};
  font-size: ${(props) => props.theme.fontSizes[props.variant]};
`

export default Text;
