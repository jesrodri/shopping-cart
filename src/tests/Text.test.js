import renderer from 'react-test-renderer';
import Text from '../components/Text';
import { ThemeProvider } from 'styled-components'
import theme from '../theme';

describe('<Text />', () => {
  it('renders Text correctly', () => {
    const text = renderer.create(
      <ThemeProvider theme={theme}>
        <Text />
      </ThemeProvider>
    ).toJSON();
    
    expect(text).toMatchSnapshot();
  });
});
