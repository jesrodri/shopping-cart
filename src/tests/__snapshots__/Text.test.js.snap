// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders Text correctly 1`] = `
<text
  className="Text-sc-jb2lm6-0 eUwnFP"
/>
`;
